plugin/evm/limit_order.go

```diff
@@ -61,105 +72,98 @@ func NewLimitOrderProcesser(ctx *snow.Context, chainConfig *params.ChainConfig,
 		backend:      backend,
 		memoryDb:     limitorders.NewInMemoryDatabase(),
 		orderBookABI: orderBookAbi,
-		blockChain: blockChain,
+		blockChain:   blockChain,
 	}
 }
 
 func (lop *limitOrderProcesser) ListenAndProcessTransactions() {
 	lastAccepted := lop.blockChain.LastAcceptedBlock().NumberU64()
 	if lastAccepted > 0 {
 		log.Info("ListenAndProcessTransactions - beginning sync", " till block number", lastAccepted)
-		
+
 		allTxs := types.Transactions{}
 		for i := uint64(0); i <= lastAccepted; i++ {
 			block := lop.blockChain.GetBlockByNumber(i)
 			if block != nil {
-				allTxs = append(allTxs, block.Transactions()...)
+				for _, tx := range block.Transactions() {
+					parseTx(lop.txPool, lop.orderBookABI, lop.memoryDb, tx, i)
+				}
 			}
 		}
-	
-		for _, tx := range allTxs {
-			parseTx(lop.orderBookABI, lop.memoryDb, *lop.backend, tx)
-		}
 
 		log.Info("ListenAndProcessTransactions - sync complete", "till block number", lastAccepted, "total transactions", len(allTxs))
 	}
 
 	lop.listenAndStoreLimitOrderTransactions()
 }
 
-func (lop *limitOrderProcesser) AddMatchingOrdersToTxPool() {
-	orders := lop.memoryDb.GetAllOrders()
-	for _, order := range orders {
-		nonce := lop.txPool.Nonce(common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")) // admin address
-
-		data, err := lop.orderBookABI.Pack("executeTestOrder", order.RawOrder, order.Signature)
-		if err != nil {
-			log.Error("abi.Pack failed", "err", err)
-		}
-		key, err := crypto.HexToECDSA("56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027") // admin private key
-		if err != nil {
-			log.Error("HexToECDSA failed", "err", err)
-		}
-		executeOrderTx := types.NewTransaction(nonce, common.HexToAddress("0x0300000000000000000000000000000000000069"), big.NewInt(0), 5000000, big.NewInt(80000000000), data)
-		signer := types.NewLondonSigner(big.NewInt(321123))
-		signedTx, err := types.SignTx(executeOrderTx, signer, key)
-		if err != nil {
-			log.Error("types.SignTx failed", "err", err)
+func (lop *limitOrderProcesser) RunMatchingEngine() {
+	purgeLocalTx(lop.txPool)
+	longOrders := lop.memoryDb.GetLongOrders()
+	shortOrders := lop.memoryDb.GetShortOrders()
+	if len(longOrders) == 0 || len(shortOrders) == 0 {
+		return
+	}
+	for _, longOrder := range longOrders {
+		var breakOuterLoop bool = false
+		for j, shortOrder := range shortOrders {
+			if longOrder.Price == shortOrder.Price && longOrder.BaseAssetQuantity == (-shortOrder.BaseAssetQuantity) {
```

Comment:
can we allow `longOrder.Price >= shortOrder.Price` ?

Reply:
lets make this change when contract changes the logic to include different price and use best price.